public/app/features/alerting/unified/components/MenuItemPauseRule.tsx

```diff
@@ -33,30 +38,36 @@ const MenuItemPauseRule = ({ rule, onPauseChange }: Props) => {
       return;
     }
     const ruleUid = rule.rulerRule.grafana_alert.uid;
+    const targetGroup = await getGroup({
+      rulerConfig: grafanaRulerConfig,
+      namespace: rule.namespace.uid || rule.rulerRule.grafana_alert.namespace_uid,
+      group: rule.group.name,
+    }).unwrap();
 
     // Parse the rules into correct format for API
-    const modifiedRules = group.rules.map((groupRule) => {
-      if (!(isGrafanaRulerRule(groupRule.rulerRule) && groupRule.rulerRule.grafana_alert.uid === ruleUid)) {
-        return groupRule.rulerRule!;
-      }
-
-      return produce(groupRule.rulerRule, (updatedGroupRule) => {
-        updatedGroupRule.grafana_alert.is_paused = newIsPaused;
-      });
-    });
+    const modifiedRules =
+      targetGroup.rules.map((groupRule) => {
+        if (!(isGrafanaRulerRule(groupRule) && groupRule.grafana_alert.uid === ruleUid)) {
+          return groupRule;
+        }
+        return produce(groupRule, (updatedGroupRule) => {
```

Comment:
non-blocking nit: Wouldn't be easier to move `produce` outside, and put the logic into to draft function, like

```
const payload =  produce(targetGroup, draftGroup => {
 const rule = group.rules.find(r => isGrafanaRulerRule(r) && r.grafana_alert.uid === ruleUid);
 rule.grafana_alert.is_paused = newIsPaused
}
```

Reply:
🤔 I'm not sure about this change. For me it's more clear the way it's already implemented, we map the rules in the `targetGroup` and in the map we use the `produce` to return the rule with only the `grafana_alert.is_paused `updated.